### PR TITLE
doc: add initial version of actions

### PR DESCRIPTION
**Pull request by mhdawson**
## Summary
Add the base actions to check PRs, build releases etc.

## Use Cases
Need so that we have published versions we  can start consuming in other
buildpacks.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by mhdawson** [comment]
@ryanmoran not sure if these is correct or not but thought I'd take a try at it since I think we need releases before I can open PRs to use the shread functionality from other packages.

It's mostly a copy of the actions/files that were there for node-engine with integration tests removed as I think those will be covered instead by the buildpacks that used the shared library. I also did not copy over the action that updated dependencies as I don't think this package has any, at least to start.

**Comment by sophiewigmore** [comment]
@mhdawson I'm not a Node maintainer, so we should wait for one of them to give you an official approval, but these changes look good to me. You probably also need to copy in the `scripts` directory, since some of the worklows use them (like the unit test step uses `./scripts/unit.sh`)

Just a heads up also - the update github config workflow will overwrite any changes to the workflows you have made when it runs (like the removal of integration tests). Our integration script should be smart enough to just warn about no integration tests, but not fail out when it runs if there aren't any tests to run.

**Comment by mhdawson** [comment]
@sophiewigmore thanks for reviewing. In terms of the scripts I think we already have those in the repo as part of the initial commit - https://github.com/paketo-buildpacks/libnodejs/tree/main/scripts

File: .github/workflows/update-github-config.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    - name: Checkout Branch
+      uses: paketo-buildpacks/github-config/actions/pull-request/checkout-branch@main
+      with:
+        branch: automation/github-config/update
+
+    - name: Run the sync action
+      uses: paketo-buildpacks/github-config/actions/sync@main
+      with:
+        workspace: /github/workspace
+        config: /github/workspace/github-config/implementation
```
Comment (ryanmoran) [inline .github/workflows/update-github-config.yml:36]:
```suggestion
        config: /github/workspace/github-config/library
```

**Comment by ryanmoran** [comment]
@mhdawson I don't think so. The key workflow here is that `update-github-config.yml` one anyway. Once that lands, the repo will converge itself on the right set of files. I'm happy to go ahead and merge this to let that process take place.

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**